builder/detection_order.go

```diff
@@ -0,0 +1,18 @@
+package builder
+
+import (
+	"github.com/buildpacks/pack/internal/dist"
+)
+
+type DetectionOrderEntry struct {
+	dist.BuildpackRef
+	Cyclical            bool           `json:"cyclic,omitempty" yaml:"cyclic,omitempty" toml:"cyclic,omitempty"`
+	GroupDetectionOrder DetectionOrder `json:"nested_buildpacks,omitempty" yaml:"nested_buildpacks,omitempty" toml:"nested_buildpacks,omitempty"`
```

Comment:
```suggestion
	GroupDetectionOrder DetectionOrder `json:"buildpacks,omitempty" yaml:"buildpacks,omitempty" toml:"buildpacks,omitempty"`
```

The term "nested" doesn't make sense when it's a top-level group. This proposed general term works well in both cases IMO.

After:

```json
{
  "detection_order": [
      {
        "buildpacks": [
          {
            "id": "samples/java-maven",
            "version": "0.0.1"
          }
        ]
      },
      {
        "buildpacks": [
          {
            "id": "samples/hello-universe",
            "version": "0.0.1",
            "buildpacks": [
              {
                "id": "samples/hello-world",
                "version": "0.0.1"
              },
              {
                "id": "samples/hello-moon",
                "version": "0.0.1"
              }
            ]
          }
        ]
      },
  ]
}
```